### PR TITLE
TOOL-7374 Changes in appliance-build to support linux-pkg build lists

### PR DESCRIPTION
**Pull request by pzakha**
## PROJECT

This change is part of the linux-pkg build list project here:
https://jira.delphix.com/browse/TOOL-7371

Note that the change will require a flag-day, see rollout strategy here:
https://docs.google.com/spreadsheets/d/1-qRT4esBl0827wKCpsNe8veXJnH2s8k1fWr20me260Q

## DESCRIPTION

appliance-build changes following TOOL-7373

## TESTING

git-ab-pre-push (this includes changes from TOOL-7374, TOOL-7375, TOOL-7376):
http://collins.d.delphix.com:21968/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/26/

### CODE REVIEW

File: live-build/build.gradle [hunk 1]
```diff
@@ -22,11 +22,11 @@ task ancillaryRepository(type: Exec) {
     inputs.file "${rootProject.projectDir}/scripts/build-ancillary-repository.sh"
 
     for (envVar in ["AWS_S3_URI_VIRTUALIZATION",
-                    "AWS_S3_URI_LINUX_PKG",
+                    "AWS_S3_URI_USERLAND_PKGS",
                     "AWS_S3_URI_MASKING",
                     "AWS_S3_URI_ZFS",
```
Comment (sebroy) [inline live-build/build.gradle]: Should this be `KERNEL`? (i.e. where is the `connstat` package coming from?)

File: live-build/build.gradle [hunk 2]
```diff
@@ -22,11 +22,11 @@ task ancillaryRepository(type: Exec) {
...
                     "AWS_S3_PREFIX_VIRTUALIZATION",
-                    "AWS_S3_PREFIX_LINUX_PKG",
+                    "AWS_S3_PREFIX_USERLAND_PKGS",
                     "AWS_S3_PREFIX_MASKING",
                     "AWS_S3_PREFIX_ZFS"]) {
```
Comment (sebroy) [inline live-build/build.gradle]: Same question here.

File: live-build/build.gradle (hunk 1 above)
Comment (pzakha) [reply live-build/build.gradle]: In the first phase of the rollout we do not use the kernel list yet, so we leave connstat and delphix-kernel as part of the userland list.
TOOL-7381 will be responsible of cleaning up the userland list.

**Comment by sebroy** [comment]
bors delegate+

**Pull request merged**